src/platform/android/java/chip/platform/AndroidBLEManager.java

```diff
@@ -237,58 +182,60 @@ public synchronized ChipDeviceController removeConnection(int connId) {
     }
   }
 
-  public static void onNotifyChipConnectionClosed(int connId) {
-    ChipDeviceController deviceController = AndroidChipStack.getInstance().getConnection(connId);
-    if (deviceController != null) {
-      deviceController.onNotifyChipConnectionClosed(connId);
+  @Override
+  public synchronized BLEConnection getConnection(int connId) {
+    int connIndex = connId - 1;
+    if (connIndex >= 0 && connIndex < mConnections.size()) {
+      return mConnections.get(connIndex);
     } else {
-      Log.i(TAG, "Tried to notify connection closed, but BLE connection was not found.");
+      Log.e(TAG, "Unknown connId " + connId);
+      return null;
     }
   }
 
-  public static boolean onSendCharacteristic(
-      int connId, byte[] svcId, byte[] charId, byte[] characteristicData) {
-    ChipDeviceController deviceController = AndroidChipStack.getInstance().getConnection(connId);
-    if (deviceController == null) {
-      Log.i(TAG, "Tried to send characteristic, but BLE connection was not found.");
-      return false;
-    }
-    BluetoothGatt bluetoothGatt = deviceController.getBluetoothGatt();
-    if (bluetoothGatt == null) {
-      return false;
-    }
+  @Override
+  public BluetoothGattCallback getCallback() {
+    return mGattCallback;
+  }
 
-    UUID svcUUID = convertBytesToUUID(svcId);
-    BluetoothGattService sendSvc = bluetoothGatt.getService(svcUUID);
-    if (sendSvc == null) {
-      Log.e(TAG, "Bad service");
-      return false;
+  private synchronized int getConnId(BluetoothGatt gatt) {
+    // Find callback given gatt
+    int connIndex = 0;
+    while (connIndex < mConnections.size()) {
+      BLEConnection connObj = mConnections.get(connIndex);
+      if (connObj != null) {
+        if (gatt == connObj.getBluetoothGatt()) {
+          return connIndex + 1;
+        }
+      }
+      connIndex++;
     }
+    return 0;
+  }
 
-    UUID charUUID = convertBytesToUUID(charId);
-    BluetoothGattCharacteristic sendChar = sendSvc.getCharacteristic(charUUID);
-    if (!sendChar.setValue(characteristicData)) {
-      Log.e(TAG, "Failed to set characteristic");
-      return false;
-    }
+  @Override
+  public int init() {
```

Comment:
It does not do anything because we have not impl it yet. Ref to src/platform/Linux/BLEManagerImpl.cpp, you will found that these function are useful later

Reply:
In that case, add a TODO?

Reply:
OK